iio/iio.c

```diff
@@ -1172,6 +1175,8 @@ static int iio_call_submit(struct iiod_ctx *ctx, const char *device,
 		return -EINVAL;
 
 	dev->buffer.public.dir = dir;
+	dev->buffer.public.cyclic_info.buff_index = 0;
```

Comment:
Hi @rbolboac,
Thanks for this PR.

I have one comment here. 
Should this line be inside the `if (dev->dev_descriptor->submit && dev->trig_idx==NO_TRIGGER)` ?
Doing this will ensure that the buff_index is being reset only when the submit function is defined and there are no triggers.

If the user makes use of trigger to push data to DAC, then resetting the cyclic buffer index to zero should be take care by the application layer or should be take care in the iio_buffer_pop scan function?

Reply:
Hi @pmallick-adi . I did not take into consideration the usage of triggers. You are right. Some adaptations have to be done for these scenario. I am currently working on implementing the needed changes.

Reply:
Thanks for considering this.